OpenRA.Mods.Common/Traits/Cloak.cs

```diff
@@ -17,6 +17,20 @@
 
 namespace OpenRA.Mods.Common.Traits
 {
+	[Flags]
+	public enum UncloakType
+	{
+		None = 0,
+		Attack = 1,
+		Move = 2,
+		Unload = 4,
+		Infiltrate = 8,
+		Demolish = 16,
+		Damage = 32,
+		[Desc("Uncloaks on Attack, Unload, Infiltrate, Demolish")]
```

Comment:
I still think we should remove this `Default` type, and make it explicit in the trait.